src/config/config_type.rs

```diff
@@ -210,7 +208,7 @@ macro_rules! create_config {
             )+
 
             #[allow(unreachable_pub)]
-            pub fn default_with_style_edition(style_edition: StyleEdition) -> Config {
+            pub(super) fn default_with_style_edition(style_edition: StyleEdition) -> Config {
```

Comment:
I may have just been caught up in the difficulty of naming things, but I did feel odd having this function as well as `default_for_possible_style_edition`. While the function names imply a lot of overlap, I do feel they are doing very different things.

This function is very much instantiating a config with the provided style edition, whereas `default_for_possible_style_edition` is more of a "figure out what style edition we should be using, and then give me the default config for that style edition"

Reply:
Might be worth adding doc comments to explain what each function's purpose is within the codebase.